### PR TITLE
Reframes the quirk point system

### PR DESCRIPTION
**Pull request by Lufferly**
## About The Pull Request

Reframes quirks, now what are "negative" quirks now give points, and "positive" quirks require points.
Renames a lot of code and some files to remove quantifiers of quirks like "negative" and "positive", focusing instead on if they cost or give points.
<details><summary>Recolors "negative" quirks on the quirk page from red to blue:</summary>
<p>

Screenshot 2023-09-12 001623

</p>
</details> 
I tested this, but its likely I missed something or fucked a boolean or whatnot, just let me know.

## Why It's Good For The Game
 
A toned down version of #78259 but with a lot of the same sentiment. Assigning negative value to real world disabilities is ableist, as is using the word "negative" to describe them. It implies that people in the real world with them have negative value, or are worth less than those without. The same reasoning is used to recolor them from red to blue, red in this context is used as a negative color, and I really dont think that we should be framing real world disabilites that way. Additionally I always kinda felt like it was backward that you spend your points to get quirks that give you points. This should make it more clear.

## Changelog

:cl: Seven
qol: Swaps the quirk point system. Now, "negative" quirks give points and "positive" quirks cost points. Additionally, "negative" quirks are now colored blue in the preference menu.
/:cl:

### CODE REVIEW

**Comment by AtomTheProphet** [comment]
This doesn't seem necessary...

**Comment by ZephyrTFA** [comment]
I don't see the point of this and makes it harder to understand without reason.

Good quirks cost points
Negative quirks reward points

**Comment by optimumtact** [comment]
How exactly does this make it harder to read?

**Comment by ZephyrTFA** [comment]
> How exactly does this make it harder to understand? It's literally just the inverse of what it was previously

It is intuitive to the average coder that if something is good, it costs X currency and if something is bad for you is gives you X currency.
I have never encountered a system where something bad costs you X currency.

**Comment by optimumtact** [comment]
if this had started the other way around you would be in the exact same PR arguing the exact same point from the other direction.

**Comment by Serxule** [comment]
i dont like it being blue for 2 reasons, 1 blue is generally seen as a "good" color like green (which is confusing if its 2 "good" colors) and its more confusing with the order being swapped, the colors adding to that problem

**Comment by Serxule** [comment]
also, you're still using the same bullshit conspiracy theory argument that "game balance of a negative impact giving negative points means irl people with disabilities are a worse person worth less" genuinely shut the fuck up, what the fuck are you talking about?

**Comment by timothymtorres** [comment]
This reminds me of when I added permanent lung damage to the smoker quirk and someone was trying to argue that smoking isn't harmful and doesn't do long term damage...

**Comment by dieamond13** [comment]
How is having a brain tumor and a chronic illness not a bad thing?

**Comment by EuSouAFazer** [comment]
I feel like this could be atomized into two PRs, one for the negative points -> positive points change and another for the color change. That way if the PR for the colors gets barred the point change still goes in

**Comment by Jezzared** [comment]
Idea: Instead of "Costs points" and "Gives points", maybe make it a "Difficulty Rating". So being blind gives you a difficulty rating of 16, but lessening that with helpful quirks lowers the difficulty rating.
Then you don't need to sort them into "Negative/Positive" or "Gives/Costs", but instead just one "Difficulty Rating". The 'error' for trying to pick a helpful one without a negative one can then be "You can't have a negative Difficulty Rating!" or something to that effect.

**Comment by Inari-Whitebear** [comment]
Not sure why you'd read it that way. It isn't a "value" judgment of your character, it's about them helping or hindering you.

"good quirks" have effects that generally help your gameplay. It has value in regards to the gameplay, not to your character's RP persona and their acceptance in society or some such.
"bad quirks" have effects that generally hinder your gameplay.

**Comment by Grayman7000** [comment]
I don't think it's ableist to acknowledge that something makes gameplay harder. Being blind is definitely negative, being deaf is definitely negative, because all that's happening is you're losing out on things with really no benefits.
I likewise agree with other people, blue is typically seen as good, my colour evil hierarchy would go green > blue > yellow > orange > red, so to contrast green and blue as the new generally beneficial and generally negative is a bad choice. I appreciate what you're doing, making the game use more inclusive language is always good, but I don't think this is the way.

**Comment by DeerJesus** [comment]
i dont think its ableist to acknowledge having all your limbs gone is not a very postitive physical trait

File: tgui/packages/tgui/interfaces/PreferencesMenu/QuirksPage.tsx [hunk 1]
```diff
@@ -143,7 +143,7 @@ export const QuirksPage = (props, context) => {
         }
 
         const {
-          max_positive_quirks: maxPositiveQuirks,
+          max_costs_points_quirks: maxtotalCostsPointsQuirks,
```
Comment (Mothblocks) [inline tgui/packages/tgui/interfaces/PreferencesMenu/QuirksPage.tsx:146]: maxTotal not maxtotal

Comment (Lufferly) [reply tgui/packages/tgui/interfaces/PreferencesMenu/QuirksPage.tsx]: done, thanks

**Comment by EricZilla** [comment]
I think this may be the first time I've ever seen literal color bikeshedding.
As someone who is actually deaf in one ear, fuck out of here with this nonsense. You are arguing not for removal of ableism, but for the addition of goodspeak. I'm not "differently abled" I'm partially disabled. I have a disability that effects me every day, every time I talk to someone. It puts me at risk while driving.
If you are a disabled person, do some growing up. If you aren't, Shut the fuck up

**Comment by Lufferly** [comment]
I do not claim that this removes ableism from the game, nor do I argue for the removal of the word disability. I simply wish to move the quirk system away from the abstraction of negative value onto disabilities. This pr does not even completely do that, disabilities are still assigned points. But I believe that this is better than how we assign negative value to these disabilities, and use negative speech wherever we refer to them. I also believe that color is an extension of that speech. Ultimately that is all this changes, the framing of these quirks, and it doesn't even completely reframe them.

**Comment by EricZilla** [comment]
I was more harsh there than I needed to be. I was about to edit an apology for my knee-jerk reaction, instead I'll put it here

**Comment by EricZilla** [comment]
I'll get into my main issue. Firstly, this is a video game so it having mechanics to reward you for making a more interesting character is good.
Secondly, it's very patronizing to say "we can't admit that lifelong struggles negatively impact people" in my opinion, obviously your heart is in the right place, and I actually prefer the 3 free good quirks system you did first. Just can't trust TG with that.

The problem I really have is, it's tone policing, and leads down the road of saying "your homophobic because you called yourself gay" which is hyperbole but it's what happens over time. Talking about social issues the "wrong way" when no harm was meant. They are just called quirks, something that makes you a little different, some in a good way some in a bad, some in a neutral way. Change the numbers sure, but is it really the end of the world to have them be red?

**Comment by Inari-Whitebear** [comment]
> I do not claim that this removes ableism from the game, nor do I argue for the removal of the word disability. I simply wish to move the quirk system away from the abstraction of negative value onto disabilities. This pr does not even completely do that, disabilities are still assigned points. But I believe that this is better than how we assign negative value to these disabilities, and use negative speech wherever we refer to them. I also believe that color is an extension of that speech. Ultimately that is all this changes, the framing of these quirks, and it doesn't even completely reframe them.

There's one thing I never quite understand about this approach though:
So.. what now? Aren't we now calling the positive quirks bad by them having negative values? Is it bad to be Jolly?

**Comment by Inari-Whitebear** [comment]
If we have to change it, I think I like the idea of changing it to a "Difficulty" wording most.

**Comment by ChungusGamer666** [comment]
Can we stop pretending that being disabled is somehow a good thing? Being deaf, being blind, and having a brain turmor are things that NEGATIVELY affect you both in real life and ingame. There is no amount of beating around the bush that will change the reality of it, these are disabilities. Changing the color to blue, which is a color generally related with "good" things, is not just absurd, it is also bad UI design.

The points change is fine and more intuitive for the player and coders, but otherwise this PR is a mess.

**Comment by optimumtact** [comment]
well i wont make you suffer through anymore

**Pull request closed without merging**